projects/packages/backup/tests/php/test-rest-controller.php

```diff
@@ -166,6 +172,219 @@ public function test_delete_backup_helper_script_success() {
 		$this->assertTrue( $response->get_data()['success'] );
 	}
 
+	/**
+	 * Testing the `/jetpack/v4/database-object/backup` endpoint with invalid params.
+	 */
+	public function test_backup_database_object_invalid_params() {
+		$request  = new WP_REST_Request( 'GET', '/jetpack/v4/database-object/backup' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 400, $response->get_status() );
+		$this->assertEquals( 'Missing parameter(s): object_type, object_id', $response->get_data()['message'] );
+
+		$request->set_query_params(
+			array(
+				'object_id'   => 123,
+				'object_type' => 'dummy',
+			)
+		);
+		$response      = $this->server->dispatch( $request );
+		$response_data = $response->get_data();
+		$this->assertEquals( 400, $response->get_status() );
+		$this->assertEquals( 'Invalid parameter(s): object_type', $response_data['message'] );
+		$this->assertEquals( 'The object_type argument should be one of woocommerce_attribute, woocommerce_downloadable_product_permission, woocommerce_order_item, woocommerce_payment_token, woocommerce_tax_rate, woocommerce_webhook', $response_data['data']['params']['object_type'] );
+
+		$request->set_query_params(
+			array(
+				'object_id'   => 'should_be_integer',
+				'object_type' => 'woocommerce_attribute',
+			)
+		);
+		$response      = $this->server->dispatch( $request );
+		$response_data = $response->get_data();
+		$this->assertEquals( 400, $response->get_status() );
+		$this->assertEquals( 'Invalid parameter(s): object_id', $response_data['message'] );
+		$this->assertEquals( 'object_id is not of type integer.', $response_data['data']['params']['object_id'] );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/database-object/backup` endpoint with admin user.
+	 */
+	public function test_backup_database_object_unauthorized() {
+		wp_set_current_user( $this->admin_id );
+
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/database-object/backup' );
+		$request->set_query_params(
+			array(
+				'object_id'   => 123,
+				'object_type' => 'woocommerce_attribute',
+			)
+		);
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 403, $response->get_status() );
+		$this->assertEquals( 'You are not allowed to perform this action.', $response->get_data()['message'] );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/database-object/backup` endpoint on success.
+	 */
+	public function test_backup_database_object_success() {
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/database-object/backup' );
+		$request->set_query_params(
+			array(
+				'object_id'   => 123,
+				'object_type' => 'woocommerce_attribute',
+			)
+		);
+		$response = $this->server->dispatch( $request );
```

Comment:
I think this `server->dispatch` call is unnecessary, since we call `dispatch_request_signed_with_blog_token` below.